### ISSUE TITLE
NetBricks integration with Tokio-rs

### ISSUE THREAD

**Issue by rohitjoshi** [NetSys/NetBricks#16]
I am looking for alternate to [Seastar] (https://github.com/scylladb/seastar) in Rust lang and seems NetBricks can be used for developing high-performance network applications.
Any plans to integrate with [tokia-rs](https://github.com/tokio-rs/tokio)?

**Comment by apanda** [comment]
Yeah, I think this is a good idea, just not sure of a timeline yet. But just to be clear, what sort of integration were you considering? What I had in mind was to eventually add in some sort of a TCP stack (Seastar has one in the `net` directory) probably building on the netbsd stack and then exposing `futures` for that. This seems to be similar to what Seastar would offer, but did you have some other use case or design in mind?

**Comment by rohitjoshi** [comment]
I am starting a new project and wanted to use Rust but did not find an alternative to Seastar.
Seems NetBricks has basic building blocks and if we add tcp stack with features, it can be alternate to seastar. so use case is same as what you describe.

**Comment by mpaxos** [comment]
I am also looking for a Seastar alternative for rust. If NetBricks can add tcp stack it would be great!
